sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java

```diff
@@ -0,0 +1,176 @@
+            .encoders(Collections.singletonList(MessageEncoder.class))
+            .decoders(Collections.singletonList(MessageDecoder.class))
+            .build();
+
+        return clientAccessUriProvider.flatMap(uri -> Mono.fromCallable(() -> {
+            this.session = clientManager.connectToServer(endpoint, config, new URI(uri));
+            return (Void) null;
+        }).subscribeOn(Schedulers.boundedElastic()));
+    }
+
+    public Mono<Void> stop() {
+        return Mono.fromCallable(() -> {
+            if (session != null && session.isOpen()) {
+                session.close(CloseReasons.NORMAL_CLOSURE.getCloseReason());
+                messageSink.tryEmitComplete();
+                messageSink = Sinks.many().multicast().onBackpressureBuffer();
+            }
+            return (Void) null;
+        }).subscribeOn(Schedulers.boundedElastic());
+    }
+
+    public Mono<WebPubSubResult> joinGroup(String group) {
+        return joinGroup(group, nextAckId());
+    }
+
+    public Mono<WebPubSubResult> joinGroup(String group, long ackId) {
+        return Mono.fromCallable(() -> {
+            session.getBasicRemote().sendObject(new JoinGroupMessage().setGroup(group).setAckId(ackId));
+            return new WebPubSubResult();
+        }).subscribeOn(Schedulers.boundedElastic()).then(waitForAckMessage(ackId));
+    }
+
+    public Mono<WebPubSubResult> leaveGroup(String group) {
+        return leaveGroup(group, nextAckId());
+    }
+
+    public Mono<WebPubSubResult> leaveGroup(String group, long ackId) {
+        return Mono.fromCallable(() -> {
+            session.getBasicRemote().sendObject(new LeaveGroupMessage().setGroup(group).setAckId(ackId));
+            return new WebPubSubResult();
+        }).subscribeOn(Schedulers.boundedElastic()).then(waitForAckMessage(ackId));
+    }
+
+    public Mono<WebPubSubResult> sendMessageToGroup(String group, BinaryData content, WebPubSubDataType dataType) {
+        return sendMessageToGroup(group, content, dataType, nextAckId(), false, false);
+    }
+
+    public Mono<WebPubSubResult> sendMessageToGroup(String group, BinaryData content, WebPubSubDataType dataType,
+                                                    long ackId, boolean noEcho, boolean fireAndForget) {
+
+        Mono<WebPubSubResult> sendMono = Mono.fromCallable(() -> {
+            BinaryData data = content;
+            if (dataType == WebPubSubDataType.BINARY) {
+                data = BinaryData.fromBytes(Base64.getEncoder().encode(content.toBytes()));
+            }
+
+            session.getBasicRemote().sendObject(new SendToGroupMessage()
+                .setGroup(group)
+                .setData(data)
+                .setDataType(dataType.name().toLowerCase(Locale.ROOT))
+                .setAckId(ackId)
+                .setNoEcho(noEcho));
+            return (WebPubSubResult) null;
+        }).subscribeOn(Schedulers.boundedElastic());
```

Comment:
The async from the websocket lib is via Future, which seems not compatible with Reactor. Hence call the sync one on boundedElastic

Reply:
Switched to use callback from websocket client (handler), and use `Mono.create` to convert the callback to `Mono`.